### PR TITLE
Make QNode max expansion accessible as a keyword argument

### PR DESCRIPTION
**Pull request by josh146**
**Context:** The maximum number of expansions allowed by a QNode is currently set to 2.

**Description of the Change:**

* The QNode has a new keyword argument, `max_expansion`, that determines the maximum number of times the internal circuit should be expanded when executed on a device. This allows it to be easily user accessible.

* The default number of max expansions has been increased to 10

**Benefits:** Devices that require multiple operation decompositions are now supported

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -31,6 +31,8 @@ jobs:
           access_token: ${{ github.token }}
 
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
```
Comment (thisac) [inline .github/workflows/tests.yml:35]: Why are more than the latest commit fetched here? Any particular reason for this change?

**Review by thisac — APPROVED** [review]
Looks godd @josh146! :100: Nice to have this parameter exposed to the user.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@
 
 <h3>Improvements</h3>
 
+* The QNode has a new keyword argument, `max_expansion`, that determines the maximum number of times
+  the internal circuit should be expanded when executed on a device.
+  [(#)](https://github.com/PennyLaneAI/pennylane/pull/)
```
Comment (thisac) [inline .github/CHANGELOG.md:13]:
```suggestion
  [(#1074)](https://github.com/PennyLaneAI/pennylane/pull/1074)
```

File: .github/workflows/tests.yml (hunk 1 above)
Comment (josh146) [reply .github/workflows/tests.yml:35]: I noticed that this was the reason codecov was failing 😆 It's why, as of today, we're no longer waiting for the codecov status to come in

Comment (thisac) [reply .github/workflows/tests.yml:35]: Oh, do you know the reason for that?

**Pull request merged**